src/components/CheckboxGroup/CheckboxGroup.test.tsx

```diff
@@ -76,18 +86,15 @@ describe("Checkbox", () => {
     expect(screen.getByLabelText("Checkbox 4")).toBeInTheDocument();
   });
 
-  it("renders with appropriate 'aria-label' attribute and value when 'showLabel' prop is set to false", () => {
+  it("<legend> element is available in the DOM when 'showLabel' prop is set to true or false", () => {
     const { rerender } = render(
       <CheckboxGroup id="checkboxGroup" labelText="Test Label" name="test2">
         <Checkbox id="checkbox2" value="2" labelText="Checkbox 2" />
         <Checkbox id="checkbox3" value="3" labelText="Checkbox 3" />
         <Checkbox id="checkbox4" value="4" labelText="Checkbox 4" />
       </CheckboxGroup>
     );
-    expect(screen.getByTestId("checkbox-group")).not.toHaveAttribute(
-      "aria-label",
-      "Test Label"
-    );
+    expect(screen.getByText("Test Label")).toBeVisible();
```

Comment:
I agree that it might be better to check differently. `.toBeVisible()` was to me a little confusing at first.